### PR TITLE
Fix macos m1 arm wheel

### PR DESCRIPTION
**Pull request by multiphaseCFD**
### Before submitting

Please complete the following checklist when submitting a PR:

- [ ] All new features must include a unit test.
      If you've fixed a bug or added code that should be tested, add a test to the
      [`tests`](../tests) directory!

- [ ] All new functions and code must be clearly commented and documented.
      If you do make documentation changes, make sure that the docs build and
      render correctly by running `make docs`.

- [ ] Ensure that the test suite passes, by running `make test`.

- [ ] Add a new entry to the `.github/CHANGELOG.md` file, summarizing the
      change, and including a link back to the PR.

- [ ] Ensure that code is properly formatted by running `make format`. 

When all the above are checked, delete everything above the dashed
line and fill in the pull request template.

------------------------------------------------------------------------------------------------------------

**Context:**

**Description of the Change:**

**Benefits:**

**Possible Drawbacks:**

**Related GitHub Issues:**

### CODE REVIEW

File: .github/workflows/wheel_macos_arm64.yml [hunk 1]
```diff
@@ -61,7 +63,7 @@ jobs:
           CIBW_ARCHS_MACOS: ${{ matrix.arch }}
 
       - uses: actions/upload-artifact@v2
-        if: ${{ github.event_name == 'release' || github.ref == 'refs/heads/master' }}
```
Comment (mlxd) [inline .github/workflows/wheel_macos_arm64.yml:64]: I think we are good, so this can be uncommented.

**Review by mlxd — COMMENTED** [review]
Great work @multiphaseCFD 
Let's revert that single line change and I'll approve

File: setup.py [hunk 2]
```diff
@@ -77,11 +77,16 @@ def build_extension(self, ext: CMakeExtension):
...
+            if os.getenv('ARCHS') == "arm64":
+                configure_args += ["-DCMAKE_CXX_FLAGS='-target arm64-apple-macos11'"]
+            else:
+                configure_args += []
             # Disable OpenMP in M1 Macs
             if os.environ.get("USE_OMP"):
                 configure_args += []
             else:
-                configure_args += ["-DENABLE_OPENMP=OFF"]
+                configure_args += ["-DENABLE_OPENMP=OFF"]
```
[the remark this answers is not in the document]
Comment (multiphaseCFD) [reply setup.py]: Thanks @mlxd , done!

**Review by mlxd — APPROVED** [review]
:100:

Happy to merge once the tests complete.

**Pull request merged**